Lares/Assets/Mechanics/Inventory/UI/InventoryUIController.cs

```diff
@@ -0,0 +1,125 @@
+using UnityEngine;
+using UnityEngine.UI;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine.Events;
+
+namespace Lares.Inventory
+{
+    public class InventoryUIController : MonoBehaviour
+    {
+        [SerializeField] private InventoryManager _inventory;
+        private List<InventoryNode> _inventoryList;
+
+        [Header("Scroll views")]
+        [SerializeField] private Transform _consumableScrollView;
+        private Transform _consumableContentHolder;
+        [SerializeField] private Transform _materialScrollView;
+        private Transform _materialContentHolder;
+        [SerializeField] private Transform _keyScrollView;
+        private Transform _keyContentHolder;
+
+        [Header ("Inventory Data")]
+        [SerializeField] private TextMeshProUGUI _itemName;
+        [SerializeField] private TextMeshProUGUI _itemDescriptionText;
+
+        [Header("Buttons")]
+        [SerializeField] private GameObject _inventoryItemButtonPrefab;
+        [SerializeField] private Button _onUseButton;
+        [SerializeField] private Button _consumableButton;
+        [SerializeField] private Button _materialsButton;
+        [SerializeField] private Button _keyButton;
+
+        private InventoryNode _currentInventoryNode = null;
+
+        private void Start()
+        {
+            InventoryItemButton.ItemSelected += SetCurrentSelected;
+            _onUseButton.onClick.AddListener(() => { CallOnUseFunction(); });
+            _consumableButton.onClick.AddListener(() => { SetToConsumableView(); });
+            _materialsButton.onClick.AddListener(() => { SetToMaterialView(); });
+            _keyButton.onClick.AddListener(() => { SetToKeyView(); });
```

Comment:
Since this is possible - add `RemoveListener()` calls in case the `InventoryUIController` gets disabled or destroyed.
```suggestion
            _onUseButton.onClick.AddListener(CallOnUseFunction);
            _consumableButton.onClick.AddListener(SetToConsumableView);
            _materialsButton.onClick.AddListener(SetToMaterialView);
            _keyButton.onClick.AddListener(SetToKeyView);
```